From 1c22fa809bb27f794c887f459561aa4c334350e1 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Wed, 13 Mar 2024 22:27:29 +0800
Subject: [PATCH 2/7] HACK: append -lzstd to linkage list

For some reason, liblldELF.a is not linked correctly with -lzstd with missing
symbols. Append -lzstd to the linker to workaround build failures.
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b1e41665c0..25a013039a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -572,7 +572,7 @@ set_target_properties(
     VS_USER_PROPS "${PROJECT_SOURCE_DIR}/cmake/VisualD.props"
 )
 # LDFLAGS should actually be in target property LINK_FLAGS, but this works, and gets around linking problems
-target_link_libraries(${LDC_LIB} ${LLVM_LIBRARIES} ${LLVM_LDFLAGS})
+target_link_libraries(${LDC_LIB} ${LLVM_LIBRARIES} ${LLVM_LDFLAGS} -lzstd)
 if(WIN32)
     target_link_libraries(${LDC_LIB} imagehlp psapi)
 elseif(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
-- 
2.45.1

